infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java

```diff
@@ -43,6 +53,59 @@ public KubernetesClientFactory(
       @Nullable @Named("che.infra.kubernetes.password") String password,
       @Nullable @Named("che.infra.kubernetes.oauth_token") String oauthToken,
       @Nullable @Named("che.infra.kubernetes.trust_certs") Boolean doTrustCerts) {
+    this.defaultConfig =
+        buildDefaultConfig(masterUrl, username, password, oauthToken, doTrustCerts);
+    this.httpClient = HttpClientUtils.createHttpClient(defaultConfig);
+  }
+
+  /**
+   * Builds the Kubernetes {@link Config} object based on a default {@link Config} object and an
+   * optional workspace Id.
+   */
+  public Config buildConfig(Config defaultConfig, @Nullable String workspaceId)
```

Comment:
Can we show that this method is not supposed to be used directly by consumers of this class by making the method protected?

Reply:
yes, I'll change it

Reply:
I fixed it commit https://github.com/eclipse/che/pull/8805/commits/2f9d7cf6e0fc28b099f36180b4c4303569e9ca43